File: fixtures/mixed/rust_and_typescript/left.ts
export function buildOrderRows(orders: Order[]): Row[] {
  const visible = orders.filter((order) => order.visible);
  const rows = visible.map((order) => ({ id: order.id, total: order.total }));
  const kept = rows.filter((row) => row.total > 0);
  return kept;
}

File: fixtures/mixed/rust_and_typescript/left.rs
pub fn build_order_rows(orders: &[Order]) -> Vec<Row> {
    let visible = orders.iter().filter(|order| order.visible).collect::<Vec<_>>();
    let rows = visible.iter().map(|order| order.to_row()).collect::<Vec<_>>();
    let kept = rows.into_iter().filter(|row| row.total > 0).collect::<Vec<_>>();
    kept
}
